### PR TITLE
GT-1558 tool settings with toggle training tips

### PR DESCRIPTION
**Pull request by levieggertcru**
This PR adds toggling of a tract's training tips from the ToolsSettings options.

One larger change I had to make was to remove the trainingTipsEnabled flag from the renderer view factories and move that to the MobileContentRenderedPageContext which get's injected into each viewModel of a rendered view when ever a new page view is navigated to or when the renderer is set.

### CODE REVIEW

File: godtools/App/Features/ToolSettings/Presentation/ToolSettings/Views/ToolSettingsOptions/ToolSettingsOptionsViewModel.swift [hunk 1]
```diff
@@ -5,8 +5,37 @@
...
+    private weak var flowDelegate: FlowDelegate?
+    
+    required init(flowDelegate: FlowDelegate, trainingTipsEnabled: Bool) {
+        
+        self.flowDelegate = flowDelegate
+        self.trainingTipsEnabled = trainingTipsEnabled
+        
+        super.init()
+        
+        trainingTipsTitle = trainingTipsEnabled ? "Hide tips" : "Training tips"
```
Comment (rachaelblue) [inline godtools/App/Features/ToolSettings/Presentation/ToolSettings/Views/ToolSettingsOptions/ToolSettingsOptionsViewModel.swift:23]: I'm assuming these will be translated later?  As far as process, do you typically build out a feature and then have a separate ticket to go back and address translations?

Comment (levieggertcru) [reply godtools/App/Features/ToolSettings/Presentation/ToolSettings/Views/ToolSettingsOptions/ToolSettingsOptionsViewModel.swift:23]: Hey @rachaelblue there hasn't been a specific process for translating.  I am planning to translate those as the final step in this task here -> https://jira.cru.org/browse/GT-1568.

File: godtools/App/Features/ToolSettings/Presentation/ToolSettings/Views/ToolSettingsOptions/BaseToolSettingsOptionsViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//
+//  Created by Levi Eggert on 5/13/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import SwiftUI
+
+class BaseToolSettingsOptionsViewModel: ObservableObject {
+    
+    @Published var trainingTipsIcon: Image = Image("")
```
[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Features/ToolSettings/Presentation/ToolSettings/Views/ToolSettingsOptions/BaseToolSettingsOptionsViewModel.swift:13]: Hey @rachaelblue I think it's definitely a good idea to have the button change upon tapping.  I'll ask Ann about the second part if it should close or not.

[the remark this answers is not in the document]
Comment (levieggertcru) [reply godtools/App/Features/ToolSettings/Presentation/ToolSettings/Views/ToolSettingsOptions/BaseToolSettingsOptionsViewModel.swift:13]: Hey @rachaelblue yep Ann said that should stay open as long as there is a visual change on the button.  Going to make these updates.

**Comment by levieggertcru** [comment]
Hey @rachaelblue I added the state change on that button and connected the icon to the view.  As far as not closing the Tool Settings I'm going to do that in another branch I'm currently working on which is branched off of this one.

**Review by rachaelblue — APPROVED** [review]
Sounds good and looks good!

**Pull request merged**